exercicios/projeto-casa/index.js

```diff
@@ -0,0 +1,165 @@
+                }
+                return res.status(400).json('Tipo de conta não pode ficar em branco')
+            }
+            return res.status(400).json('Data de nascimento do cliente não pode ficar em branco')
+        }
+        return res.status(400).json('CPF do cliente não pode ficar em branco')
+    }
+    return res.status(400).json('Nome do cliente não pode ficar em branco')
+})
+
+// Conseguir Filtrar os clientes do banco pelo seu nome,por saldo... - DONE?
+
+app.get ('/cliente', (req, res) => {
+    const nomeFiltro = req.query.nome
+//  const criacaoFiltro = req.query.criacao - não sei como ler data
+    const contaFiltro = req.query.conta
+    const saldoFiltro = req.query.saldo
+//  const nascFiltro = req.query.nascimento - tbm nao sei ler essa data
+    const cpfFiltro = req.query.cpf
+
+    const clientesFiltrados = listaClientes.filter ((cliente, index) => {
+        if (nomeFiltro){
+            if (cliente.nome_cliente.toLowerCase().includes(nomeFiltro.toLowerCase())) return cliente
+        }
+        if (contaFiltro){
+            if (cliente.conta.tipo.toLowerCase().includes(contaFiltro.toLowerCase())) return cliente
+        }
+        if (saldoFiltro){
+            if (cliente.conta.saldo <= saldoFiltro) return cliente
+        }
+        if (cpfFiltro){
+            if (cliente.conta.cpf_cliente == cpfFiltro) return cliente
+        }
+        
+    })
+    if (clientesFiltrados.length == 0){
+        return res.status(404).json({message: "Nenhum cliente encontrado com o filtro usado. Para a lista completa, use o endereço: '/clientes'"})
+    }
+    else {
+        return res.json(clientesFiltrados)
+    }
+})   
+
+// Atualizar informações desses clientes ( como endereço, telefone de contato...) - DONE
+
+app.patch ('/cliente/:id/atualizar', (req, res) => {
+    const clienteID = req.params.id
+    const dadosAtualizados = req.body
+
+    const clienteExiste = listaClientes.find ( cliente => cliente.id == clienteID)
+    if (clienteExiste){
+        listaClientes.map ((cliente, index) => {
+            if (cliente.id == clienteID){
+                const infosAtualizadas = {
+                    ...clienteExiste, ...dadosAtualizados
+                }
+                listaClientes[index] = infosAtualizadas
+            }
+        })
+        return res.status(202).json({message:`Dados do cliente atualizados com sucesso`})
+    }
+    return res.status(404).json({message:"Cliente não encontrado"})
+})
+
+// Fazer depósitos / pagamentos usando o saldo de sua conta - DONE
+
+app.patch ('/cliente/:id/saldo', (req, res) => {
+    const clienteID = req.params.id
+    const { operacao } = req.body
+
+    const contaExiste = listaClientes.find((conta, index) => {
+        if (conta.id == clienteID){
+            return conta
+        }
+    })
+    if (contaExiste){
+        listaClientes.map ((conta, index) => {
+            if (conta.id == clienteID){
+                if (operacao >= 0){
```

Comment:
Se o valor da operação for maior que 0 = permito que o valor seja adicionado ao saldo...

Reply:
Não entendi o comentario...
É isso q a condição ta fazendo...

Reply:
Temos uma condição para valores positivos e negativos. Se esse for um caso , não ficou muito claro!